hbase/src/main/java/org/apache/metamodel/hbase/HBaseUpdateCallback.java

```diff
@@ -0,0 +1,137 @@
+
+    private final HBaseConfiguration _configuration;
+
+    private final HBaseDataContext _dataContext;
+
+    public HBaseUpdateCallback(HBaseDataContext dataContext) {
+        super(dataContext);
+        _configuration = dataContext.getConfiguration();
+        _dataContext = dataContext;
+    }
+
+    @Override
+    public TableCreationBuilder createTable(Schema schema, String name) throws IllegalArgumentException,
+            IllegalStateException {
+        throw new UnsupportedOperationException(
+                "Use createTable(Schema schema, String name, HBaseColumn[] outputColumns)");
+    }
+
+    public HBaseCreateTableBuilder createTable(Schema schema, String name, HBaseColumn[] outputColumns)
+            throws IllegalArgumentException, IllegalStateException {
+        return new HBaseCreateTableBuilder(this, schema, name, outputColumns);
+    }
+
+    @Override
+    public boolean isDropTableSupported() {
+        return true;
+    }
+
+    @Override
+    public TableDropBuilder dropTable(Table table) throws IllegalArgumentException, IllegalStateException,
+            UnsupportedOperationException {
+        return new HBaseTableDropBuilder(table, this);
+    }
+
+    public void dropTableExecute(Table table) {
+        try {
+            final HBaseWriter HbaseWriter = new HBaseWriter(HBaseDataContext.createConfig(_configuration));
+            HbaseWriter.dropTable(table.getName());
+            MutableSchema schema = (MutableSchema) table.getSchema();
+            schema.removeTable(table);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public RowInsertionBuilder insertInto(Table table) throws IllegalArgumentException, IllegalStateException,
+            UnsupportedOperationException {
+        throw new UnsupportedOperationException("Use insertInto(Table table, HBaseColumn[] outputColumns)");
+    }
+
+    public HBaseRowInsertionBuilder insertInto(Table table, HBaseColumn[] columns) throws IllegalArgumentException {
+        if (table instanceof HBaseTable) {
+            return new HBaseRowInsertionBuilder(this, (HBaseTable) table, columns);
+        }
+        throw new IllegalArgumentException("Not an HBase table: " + table);
+    }
+
+    protected synchronized void writeRow(HBaseTable hBaseTable, HBaseColumn[] outputColumns, Object[] values) {
+        try {
+            final HBaseWriter HbaseWriter = new HBaseWriter(HBaseDataContext.createConfig(_configuration));
+            HbaseWriter.writeRow(hBaseTable, outputColumns, values);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public boolean isDeleteSupported() {
+        return false;
+    }
+
+    @Override
+    public RowDeletionBuilder deleteFrom(Table table) throws IllegalArgumentException, IllegalStateException,
+            UnsupportedOperationException {
```

Comment:
Remove the declarations of thrown exceptions 'java.lang.IllegalArgumentException', 'java.lang.IllegalStateException' and 'java.lang.UnsupportedOperationException' which are runtime exceptions.